pkg/api/handlers/libpod/containers_create.go

```diff
@@ -39,6 +42,29 @@ func CreateContainer(w http.ResponseWriter, r *http.Request) {
 		utils.Error(w, http.StatusInternalServerError, fmt.Errorf("decode(): %w", err))
 		return
 	}
+
+	newImage, resolvedName, err := runtime.LibimageRuntime().LookupImage(sg.Image, &libimage.LookupImageOptions{})
+	if err != nil {
+		if errors.Is(err, storage.ErrImageUnknown) {
+			utils.Error(w, http.StatusNotFound, fmt.Errorf("no such image: %w", err))
+			return
+		}
+
+		utils.Error(w, http.StatusInternalServerError, fmt.Errorf("looking up image: %w", err))
+		return
+	}
+	imgNameOrID := newImage.ID()
+	// if the img had multi names with the same sha256 ID, should use the InputName, not the ID
+	if len(newImage.Names()) > 1 {
```

Comment:
This seems to be working around something I don't yet understand.  If `sg.Image` resolves to an image, we can just continue using that value.

But I am trying to understand `if the img had multi names with the same sha256 ID, should use the InputName, not the ID`.  Why does an image with more than one repo/tag matter?  And why are we only calling `IsRegistryReference` in that case?  Do we need to call it at all?

Reply:
I was following what the compat API was doing, but fixed now to check for the error returned from MakeContainer instead.